src/main/java/parser/CommandParser.java

```diff
@@ -0,0 +1,61 @@
+package parser;
+import commands.AddDeadlineCommand;
+import commands.AddEventCommand;
+import commands.AddToDoCommand;
+import commands.Command;
+import commands.DeleteCommand;
+import commands.FindCommand;
+import commands.InvalidCommand;
+import commands.ListCommand;
+import commands.MarkDoneCommand;
+import exceptions.MorganException;
+
+
+/**
+ * This is an parser.CommandParser class, which translates user input
+ * into commands.
+ */
+public class CommandParser {
+    private static final int COMMAND_INDEX = 0;
+    private static final String DELIMITER = "\\s+";
+    private static final String DELIMITER_FOUND_ERROR_MESSAGE = "Please "
+            + "avoid using the symbol \"" + TaskParser.DELIMITER + "\".";
+
+    /**
+     * Returns the command specified by the user.
+     * @param userInput The input string entered by the user.
+     * @return The command specified by the user.
+     * @throws MorganException If input format is invalid.
+     */
+    public Command getCommand(String userInput) throws MorganException {
+        // Checks if user input contains delimiter.
+        if (userInput.contains(TaskParser.DELIMITER)) {
+            throw new MorganException(DELIMITER_FOUND_ERROR_MESSAGE);
+        }
+
+        // Retrieve relevant data from user input
+        String trimmedUserInput = userInput.trim();
+        String[] words = trimmedUserInput.split(DELIMITER);
+        String commandKeyword = words[COMMAND_INDEX].toLowerCase();
+
+        switch (commandKeyword) {
+        case FindCommand.KEYWORD:
```

Comment:
Perhaps you can consider having a default Enum/Case, and have one of the switch statements redirect to it?